tests/components/vera/common.py

```diff
@@ -1,65 +1,123 @@
-        self.init_controller_mock = init_controller_mock
+    def __init__(self, vera_controller_class_mock):
+        """Initialize the factory."""
+        self.vera_controller_class_mock = vera_controller_class_mock
 
     async def configure_component(
-        self,
-        hass: HomeAssistant,
-        devices: Tuple[VeraDevice] = (),
-        scenes: Tuple[VeraScene] = (),
-        setup_callback: Callable[[VeraController], None] = None,
+        self, hass: HomeAssistant, controller_config: ControllerConfig
     ) -> ComponentData:
         """Configure the component with specific mock data."""
-        controller_url = "http://127.0.0.1:123"
-
-        hass_config = {
-            DOMAIN: {CONF_CONTROLLER: controller_url},
+        component_config = {
+            **(controller_config.config or {}),
+            **(controller_config.options or {}),
         }
 
-        controller = MagicMock(spec=VeraController)  # type: VeraController
-        controller.base_url = controller_url
+        controller = MagicMock(spec=pv.VeraController)  # type: pv.VeraController
+        controller.base_url = component_config.get(CONF_CONTROLLER)
         controller.register = MagicMock()
-        controller.get_devices = MagicMock(return_value=devices or ())
-        controller.get_scenes = MagicMock(return_value=scenes or ())
+        controller.start = MagicMock()
+        controller.stop = MagicMock()
+        controller.refresh_data = MagicMock()
+        controller.temperature_units = "C"
+        controller.serial_number = controller_config.serial_number
+        controller.get_devices = MagicMock(return_value=controller_config.devices)
+        controller.get_scenes = MagicMock(return_value=controller_config.scenes)
 
         for vera_obj in controller.get_devices() + controller.get_scenes():
             vera_obj.vera_controller = controller
 
         controller.get_devices.reset_mock()
         controller.get_scenes.reset_mock()
 
-        if setup_callback:
-            setup_callback(controller, hass_config)
+        if controller_config.setup_callback:
+            controller_config.setup_callback(controller)
+
+        self.vera_controller_class_mock.return_value = controller
 
-        def init_controller(base_url: str) -> list:
-            nonlocal controller
-            return [controller, True]
+        hass_config = {}
 
-        self.init_controller_mock.side_effect = init_controller
+        # Setup component through config file import.
+        if controller_config.config_from_file:
+            hass_config[DOMAIN] = component_config
 
         # Setup Home Assistant.
         assert await async_setup_component(hass, DOMAIN, hass_config)
         await hass.async_block_till_done()
 
-        return ComponentData(controller=controller)
+        # Setup component through config flow.
+        if not controller_config.config_from_file:
+            await hass.config_entries.flow.async_init(
```

Comment:
If we just want to set up the integration we can skip the config flow, add a mock config entry and use `hass.config_entries.async_setup` to set the config entry up.